data/src/main/java/org/apache/iceberg/data/RegistryBasedFileWriterFactory.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.data;
```

Comment:
Why is this located in the data package? Shouldn't this be in core?

Reply:
That's a good question. @danielcweeks  suggested to move this stuff to the data package as it is what deals with reading and writing generic content files. If we rebrand `data` as a package for accessing content files, then I think it is a good place for it. Also this helps to remove some weight from the `core` module which is getting quite big. (I have very bad experience with the Hive `ql` module which after 10 years contained everything)

Reply:
Moved the interface methods, and this factory to the `core` module